File: typespec/common/candidacies.go
package common

import (
	"time"
)

type GetCandidacyInfoRequest struct {
	CandidacyID string `json:"candidacy_id"`
}

type GetCandidacyCommentsRequest struct {
	CandidacyID string `json:"candidacy_id"`
}

type CommenterType string

const (
	CommenterTypeOrgUser CommenterType = "ORG_USER"
	CommenterTypeHubUser CommenterType = "HUB_USER"
)

type CandidacyComment struct {
	CommentID     string        `json:"comment_id"`
	CommenterName string        `json:"commenter_name"`
	CommenterType CommenterType `json:"commenter_type"`
	Content       string        `json:"content"`
	CreatedAt     time.Time     `json:"created_at"`
}

File: typespec/common/candidacies.ts
import { CandidacyState } from "./interviews";

export interface GetCandidacyInfoRequest {
  candidacy_id: string;
}

export interface GetCandidacyCommentsRequest {
  candidacy_id: string;
}

export type CommenterType = "ORG_USER" | "HUB_USER";

export const CommenterTypes = {
  ORG_USER: "ORG_USER" as CommenterType,
  HUB_USER: "HUB_USER" as CommenterType,
};

export interface CandidacyComment {
  comment_id: string;
  commenter_name: string;
  commenter_type: CommenterType;
  content: string;
  created_at: Date;
}
